tests/test_assembleCoadd.py

```diff
@@ -0,0 +1,261 @@
+        AssembleCoaddConfig.setDefaults(self)
+        self.doWrite = False
+
+
+class MockAssembleCoaddTask(AssembleCoaddTask):
+
+    """Lightly modified version of `AssembleCoaddTask` for use with unit tests.
+    """
+    ConfigClass = MockAssembleCoaddConfig
+
+    def __init__(self, **kwargs):
+        CoaddBaseTask.__init__(self, **kwargs)
+        self.warpType = self.config.warpType
+        self.makeSubtask("interpImage")
+        self.makeSubtask("scaleZeroPoint")
+
+    def processResults(self, *args, **kwargs):
+        "This should be tested separately."
+        pass
+
+    def _dataRef2DebugPath(self, *args, **kwargs):
+        raise NotImplementedError("This lightweight version of the task is not "
+                                  "meant to test debugging options.")
+
+    def runQuantum(self, mockSkyInfo, mockRefList, *args):
+        """Simplified API for testing image coaddition algorithms.
+
+        Parameters
+        ----------
+        mockSkyInfo : `lsst.pipe.base.Struct`
+            A simple container that supplies a bounding box and WCS in the
+            same format as the output of
+            `lsst.pipe.tasks.CoaddBaseTask.getSkyInfo`
+        mockRefList : `list` of `lsst.pipe.tasks.MockGen2ExposureReference`
+            Description
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            The coadded exposure and associated metadata.
+        """
+        inputs = self.prepareInputs(mockRefList)
+        supplementaryData = self.makeSupplementaryData(mockSkyInfo, warpRefList=inputs.tempExpRefList)
+
+        retStruct = self.run(mockSkyInfo, inputs.tempExpRefList, inputs.imageScalerList,
+                             inputs.weightList, supplementaryData=supplementaryData)
+        return retStruct
+
+    def runDataRef(self, mockSkyInfo, selectDataList=None, warpRefList=None):
+        """Simplified API for testing image coaddition algorithms.
+
+        Notes
+        -----
+        This tests the Gen 2 Butler interface, and can be removed once that
+        is fully deprecated.
+
+        Parameters
+        ----------
+        mockSkyInfo : `lsst.pipe.base.Struct`
+            A simple container that supplies a bounding box and WCS in the
+            same format as the output of
+            `lsst.pipe.tasks.CoaddBaseTask.getSkyInfo`
+        warpRefList : `list` of `lsst.pipe.tasks.MockExposureReference`
+            Description
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            The coadded exposure and associated metadata.
+        """
+        inputData = self.prepareInputs(warpRefList)
+        supplementaryData = self.makeSupplementaryData(mockSkyInfo, warpRefList=inputData.tempExpRefList)
+        retStruct = self.run(mockSkyInfo, inputData.tempExpRefList, inputData.imageScalerList,
+                             inputData.weightList, supplementaryData=supplementaryData)
+        return retStruct
```

Comment:
That is correct: all of the Gen 2 vs Gen 3 differences are in the way the data references are handled by subroutines. These simple `runDataRef` and `runQuantum` methods are needed in order to replace the real methods when `makeSupplementaryData` is called, since it calls `runDataRef` (or `runQuantum` for Gen  3).